crates/hyperdrive-math/src/long/open.rs

```diff
@@ -58,7 +61,83 @@ impl State {
     ) -> FixedPoint {
         let mut state: State = self.clone();
         state.info.bond_reserves -= bond_amount.into();
-        state.info.share_reserves += (base_amount / state.vault_share_price()).into();
+        state.info.share_reserves += (base_amount / state.vault_share_price()
+            - self.open_long_governance_fee(base_amount) / state.vault_share_price())
+        .into();
         state.get_spot_price()
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use eyre::Result;
+    use fixed_point_macros::fixed;
+    use rand::{thread_rng, Rng};
+    use test_utils::{
+        agent::Agent,
+        chain::{Chain, TestChain},
+        constants::FUZZ_RUNS,
+    };
+
+    use super::*;
+
+    #[tokio::test]
+    async fn fuzz_get_spot_price_after_long() -> Result<()> {
+        // Spawn a test chain and create two agents -- Alice and Bob. Alice
+        // is funded with a large amount of capital so that she can initialize
+        // the pool. Bob is funded with a small amount of capital so that we
+        // can test `get_max_short` when budget is the primary constraint.
```

Comment:
last sentence here looks like copy pasta from somewhere else -- no max short in this test right